lib/core/application/payment/payment_bloc/payment_bloc.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:app/core/domain/event/entities/event_tickets_pricing_info.dart';
+import 'package:app/core/domain/event/input/calculate_tickets_pricing_input/calculate_tickets_pricing_input.dart';
+import 'package:app/core/domain/event/repository/event_ticket_repository.dart';
+import 'package:app/core/domain/payment/entities/payment_card_entity/payment_card_entity.dart';
+import 'package:app/core/domain/payment/payment_repository.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'payment_bloc.freezed.dart';
+
+part 'payment_state.dart';
+
+class PaymentBloc extends Cubit<PaymentState> {
+  PaymentBloc(
+    this._repository,
+    this._eventTicketRepository,
+  ) : super(PaymentState.initial());
+
+  final PaymentRepository _repository;
+  final EventTicketRepository _eventTicketRepository;
+
+  Future<void> initializeStripePayment() async {
+    final result = await _repository.getPublishableKey();
+    result.fold((l) {}, (publishableKey) {
+      emit(
+        state.copyWith(
+          status: PaymentStatus.initial,
+          publishableKey: publishableKey,
+        ),
+      );
+    });
+  }
+
+  Future<void> getListCard() async {
+    emit(state.copyWith(status: PaymentStatus.loading));
+    final result = await _repository.getListCard();
+    result.fold((l) {}, (listCard) {
+      emit(
+        state.copyWith(
+          status: PaymentStatus.loaded,
+          selectedCard: listCard.isEmpty ? null : listCard[0],
+          listCard: listCard,
+        ),
+      );
+    });
+  }
+
+  Future<void> calculatePricing({
+    required CalculateTicketsPricingInput input,
```

Comment:
I think you can separate between
- StripePaymentBloc - which only handle payment 
- EventOrderBloc (or whatever u wanna name it) - which handle interact with our system api (calculatePricing, event order related stuff)

Reply:
Stripe is just another package that supports our process, think of it as image_picker. All payment handling must go through BE server with support of flutter_stripe package